src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerExecDecorator.java

```diff
@@ -100,198 +97,222 @@ public ContainerExecDecorator(KubernetesClient client, String podName, String co
         this(client, podName, containerName, null);
     }
 
-    @Override
-    public Launcher decorate(final Launcher launcher, final Node node) {
-        return new Launcher.DecoratedLauncher(launcher) {
-            @Override
-            public Proc launch(ProcStarter starter) throws IOException {
-                if (!waitUntilContainerIsReady()) {
-                    throw new IOException("Failed to execute shell script inside container " +
-                            "[" + containerName + "] of pod [" + podName + "]." +
-                            " Timed out waiting for container to become ready!");
-                }
+    private class ClosableLauncher extends Launcher.DecoratedLauncher implements Closeable {
```

Comment:
Do we really need to create a new class? Can't we just have the exiting class add things to a list of closeables?